### PR TITLE
Add Triton Backend for Unlimited-OCR R-SWA

### PR DESCRIPTION
**Pull request by andakai**
## Summary

This PR adds explicit TritonAttention support for Unlimited-OCR's R-SWA decode
mask. The Triton path implements the mask with:

```python
if USE_R_SWA:
    prefix_len = tl.load(rswa_prefix_lens_ptr + seq_idx)
    in_prefix = seq_offset[None, :] < prefix_len
    in_window = (query_abs_pos - seq_offset) < R_SWA_WINDOW
    seq_mask = seq_mask & (in_prefix | in_window)
```

This keeps prompt/image prefix tokens globally visible, while generated tokens
use a causal sliding window. The implementation reuses existing
`rswa_prefix_lens` metadata and applies the mask inside Triton unified
attention. It also allows the R-SWA path to use TritonAttention's 3D segmented
softmax decode path when applicable.

## Test Environment

```text
GPU: 8 x NVIDIA H20-3e
GPU architecture: Hopper SM90, compute capability 9.0
Driver: 580.105.08
CUDA runtime reported by nvidia-smi: 13.0
Model: baidu/Unlimited-OCR
Dataset: OmniDocBench v1.6
```

## Performance

20-sample OmniDocBench v1.6 run:

| Backend | Elapsed | Completion tok/s | Speedup vs Flex |
|---|---:|---:|---:|
| FlexAttention | 97.38s | 248.74 | 1.00x |
| Triton R-SWA 2D | 49.78s | 485.83 | 1.96x |
| Triton R-SWA 3D | 35.28s | 692.12 | 2.76x |

Full OmniDocBench v1.6 8GPU run:

| Backend | e2e time | Pages/hour | Completion tok/s |
|---|---:|---:|---:|
| FlexAttention | 1565.34s | 3679.72 | 1341.50 |
| Triton R-SWA 3D | 463.67s | 12818.57 | 4980.91 |

Full-run speedup:

```text
3.38x faster e2e time
3.48x higher pages/hour
3.71x higher completion tok/s
```

## Accuracy

20-sample OmniDocBench v1.6 no-CDM metrics:

| Metric | FlexAttention | Triton R-SWA 3D |
|---|---:|---:|
| Text Edit ↓ | 0.123511843 | 0.121357226 |
| Formula Edit ↓ | 0.201904011 | 0.200388178 |
| Table TEDS ↑ | 87.768167687 | 87.768167687 |
| Table TEDS-S ↑ | 100.000000000 | 100.000000000 |
| Read-order Edit ↓ | 0.103801314 | 0.103801314 |

Full OmniDocBench v1.6 no-CDM metrics:

| Metric | FlexAttention | Triton R-SWA 3D |
|---|---:|---:|
| Text Edit ↓ | 0.09199 | 0.09016 |
| Formula Edit ↓ | 0.10277 | 0.10061 |
| Table TEDS ↑ | 89.73 | 90.06 |
| Table TEDS-S ↑ | 92.98 | 93.37 |
| Read-order Edit ↓ | 0.14807 | 0.14684 |

The Triton backend is slightly better than the Flex baseline on all metrics in this run.

### 20-Sample Test

Start a Triton server:

```bash
VLLM_REPO=<path-to-vllm-unlimited-ocr-repo>
VLLM_PYTHON=<path-to-vllm-python>
MODEL_PATH=<path-to-baidu-Unlimited-OCR>
OMNIDOCBENCH_IMAGES=<path-to-OmniDocBench-images>

PYTHONPATH=${VLLM_REPO} \
HF_HUB_OFFLINE=1 TRANSFORMERS_OFFLINE=1 \
${VLLM_PYTHON} -m vllm.entrypoints.cli.main serve \
  ${MODEL_PATH} \
  --served-model-name baidu/Unlimited-OCR \
  --host 127.0.0.1 \
  --port 18121 \
  --max-model-len 32768 \
  --limit-mm-per-prompt '{"image":1}' \
  --allowed-local-media-path ${OMNIDOCBENCH_IMAGES} \
  --logits-processors vllm.model_executor.models.deepseek_ocr:NGramPerReqLogitsProcessor \
  --attention-config '{"backend": "TRITON_ATTN"}'
```

Run the first 20 OmniDocBench samples:

```bash
VLLM_PYTHON=<path-to-vllm-python>
OMNIDOCBENCH_JSON=<path-to-OmniDocBench.json>
OMNIDOCBENCH_IMAGES=<path-to-OmniDocBench-images>
RUN_OMNIDOCBENCH_SCRIPT=<path-to-run_vllm_omnidocbench.py>
OUTPUT_DIR=<path-to-20-sample-output-dir>

${VLLM_PYTHON} \
  ${RUN_OMNIDOCBENCH_SCRIPT} \
  --dataset ${OMNIDOCBENCH_JSON} \
  --images-dir ${OMNIDOCBENCH_IMAGES} \
  --out-dir ${OUTPUT_DIR} \
  --server http://127.0.0.1:18121 \
  --model baidu/Unlimited-OCR \
  --prompt 'document parsing.' \
  --image-mode gundam \
  --max-tokens 8192 \
  --ngram-size 35 \
  --ngram-window 128 \
  --limit 20
```

### CODE REVIEW

File: vllm/model_executor/models/config.py
Comment (Isotr0py) [inline vllm/model_executor/models/config.py:103]: Can you also update this? Otherwise we won't fallback to triton implementation

File: vllm/model_executor/models/config.py [hunk 1]
```diff
@@ -140,11 +143,16 @@ def verify_and_update_config(vllm_config: "VllmConfig") -> None:
...
+            logger.info(
+                "Unlimited-OCR: TritonAttention — R-SWA via unified attention mask."
+            )
+
         else:
             raise ValueError(
                 f"Unlimited-OCR: unsupported attention backend "
                 f"{attn_config.backend!r} for R-SWA. "
-                "Use FLASH_ATTN (FA4) or FLEX_ATTENTION."
+                "Use FLASH_ATTN (FA4), FLEX_ATTENTION, or TRITON_ATTN."
```
Comment (Isotr0py) [inline vllm/model_executor/models/config.py]:
```suggestion
                "Use FLASH_ATTN (FA4), TRITON_ATTN or FLEX_ATTENTION."
```
Nit

File: vllm/model_executor/models/config.py [hunk 2]
```diff
@@ -140,11 +143,16 @@ def verify_and_update_config(vllm_config: "VllmConfig") -> None:
                 ),
             )
 
+        elif attn_config.backend == AttentionBackendEnum.TRITON_ATTN:
+            logger.info(
+                "Unlimited-OCR: TritonAttention — R-SWA via unified attention mask."
+            )
+
```
Comment (Isotr0py) [inline vllm/model_executor/models/config.py]: Let's move triton backend condition above flex attention.

**Comment by andakai** [comment]
Thanks for the comments! All resovled now.

**Review by Isotr0py — APPROVED** [review]

**Comment by andakai** [comment]
There seems to be an unrelated error in the ci? @Isotr0py could you take a look?
